src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -63,9 +63,21 @@ public void TestRootGetProcessById()
             Assert.Equal(1, p.Id);
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void ProcessStart_UseShellExecute_OnLinux_ThrowsIfNoProgramInstalled()
+        {
+            List<string> allowedProgramsToRun = GetAllowedProgramsToRun();
+            if (!allowedProgramsToRun.Any(program => IsProgramInstalled(program)))
+            {
+                Console.WriteLine($"None of the following programs were installed on this machine: {string.Join(",", allowedProgramsToRun)}.");
+                Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = Environment.CurrentDirectory }));
+            }
+        }
+
         [Theory, InlineData(true), InlineData(false)]
         [OuterLoop("Opens program")]
-        public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool isFolder)
+        public void ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled(bool isFolder)
```

Comment:
this is named `ProcessStart_UseShellExecute_OnUnix_SuccessWhenProgramInstalled` but it is not marked to be unix only (and reference to Windows in the code below)